src/gmt_plot.c

```diff
@@ -10833,12 +10844,26 @@ void gmt_plot_image_graticules (struct GMT_CTRL *GMT, struct GMT_GRID *G, struct
 		else if (intensity)
 			gmt_illuminate (GMT, *intensity, fill.rgb);
 		n = gmt_graticule_path (GMT, &xx, &yy, 1, true, G->x[col] - inc2[GMT_X], G->x[col] + inc2[GMT_X], G->y[row] - inc2[GMT_Y], G->y[row] + inc2[GMT_Y]);
-		gmt_setfill (GMT, &fill, pen != NULL);
+		gmt_setfill (GMT, &fill, outline);
 		S->data[GMT_X] = xx;    S->data[GMT_Y] = yy;    S->n_rows = n;
 		gmt_geo_polygons (GMT, S);
 		gmt_M_free (GMT, xx);
 		gmt_M_free (GMT, yy);
-    }
+	}
+	if (delay_outline) {	/* Just get graticule outlines and draw them */
+ 		gmt_setpen (GMT, pen);	/* Set outline pen */
+		gmt_setfill (GMT, NULL, 1);	/* Turn off fill */
+	 	gmt_M_grd_loop (GMT, G, row, col, ij) { /* Visit each node */
+			if (gmt_M_is_fnan (G->data[ij]) && skip) continue;
+			if (I && skip && gmt_M_is_fnan (I->data[ij])) continue;
+			n = gmt_graticule_path (GMT, &xx, &yy, 1, true, G->x[col] - inc2[GMT_X], G->x[col] + inc2[GMT_X], G->y[row] - inc2[GMT_Y], G->y[row] + inc2[GMT_Y]);
+			S->data[GMT_X] = xx;    S->data[GMT_Y] = yy;    S->n_rows = n;
+			gmt_geo_polygons (GMT, S);
+			gmt_M_free (GMT, xx);
```

Comment:
Can't those guys be freed **outside** the loop?

Reply:
No, unless you rewrite it. _n_ can vary from a few to tens or more depending on projection and **MAP_LINE_STEP**.  I did  not notice that it was particularly slow?